primap2/pm2io/_GHG_inventory_reading.py

```diff
@@ -34,11 +38,16 @@ def nir_add_unit_information(
     unit_row : str or int
         String "header" to indicate that the column header should be used to derive the
         unit information or an integer specifying the row to use for unit information.
+        If entity and unit information are given in the same row use only unit_row.
+    entity_row : int
+        integer specifying the row to use for entity information.
+        If entity and unit information are given in the same row use only unit_row
     unit_info : dict
         A dict with the fields
         * regexp_entity: regular expression that extracts the entity from the cell value
         * regexp_unit: regular expression that extracts the unit from the cell value
-        * manual_repl: optional: dict defining unit and entity for given cell values
+        * manual_repl_unit: optional: dict defining unit for given cell values
+        * manual_repl_entity: optional: dict defining entity for given cell values
         * default_unit: unit to be used if no unit can be extracted an no unit is given
```

Comment:
It would be much better to have all these as separate parameters. There is no reason to hide them in a dict (it is not passed around further) and using separate parameters, we can use type hints etc.

Reply:
OK, can be done. But I'll curse you when I have to adapt all BUR reading scripts.

Reply:
Now implemented